# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/QuicSocketControl.cpp
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e netwerk/protocol/http/QuicSocketControl.cpp
--- a/netwerk/protocol/http/QuicSocketControl.cpp	Mon Oct 21 19:33:02 2019 +0000
+++ b/netwerk/protocol/http/QuicSocketControl.cpp	Mon Oct 21 19:49:01 2019 +0000
@@ -64,16 +64,11 @@
   uint32_t state = nsIWebProgressListener::STATE_IS_SECURE;
 
   bool distrustImminent;
-  nsCOMPtr<nsIX509CertList> succeededCertChain;
-  nsresult rv = TransportSecurityInfo::ConvertCertArrayToCertList(
-      mSucceededCertChain, getter_AddRefs(succeededCertChain));
 
-  nsresult srv;
-  if (NS_SUCCEEDED(rv)) {
-    srv = IsCertificateDistrustImminent(succeededCertChain, distrustImminent);
-  }
+  nsresult rv =
+      IsCertificateDistrustImminent(mSucceededCertChain, distrustImminent);
 
-  if (NS_SUCCEEDED(rv) && NS_SUCCEEDED(srv) && distrustImminent) {
+  if (NS_SUCCEEDED(rv) && distrustImminent) {
     state |= nsIWebProgressListener::STATE_CERT_DISTRUST_IMMINENT;
   }
 